internal/controllers/gateway/httproute_controller.go

```diff
@@ -699,3 +751,74 @@ func updateAcceptedConditionInRouteParentStatus(
 	}
 	return changed
 }
+
+// TODO: extract method for different types of *Routes to update conditions.
+func (r *HTTPRouteReconciler) ensureParentsProgrammedCondition(
```

Comment:
There is an `ensureParentsProgrammedCondition` method for each route type that look quite similar to each other. Is there any way to use the same function for all of them?

Reply:
will split out an issue to extract `ensure*Condition` to reduce the duplicate code.

Reply:
Did you already create such an issue? I'm afraid it will pass under the radar otherwise

Reply:
See: #3390